test_util/cluster_api.py

```diff
@@ -271,69 +253,55 @@ def __init__(self, dcos_url, masters, public_masters, slaves, public_slaves,
         self.zk_hostports = ','.join(':'.join([host, '2181']) for host in self.public_masters)
         self.dns_search_set = dns_search_set
         self.provider = provider
-        self.auth_enabled = auth_enabled
         self.default_os_user = default_os_user
         self.web_auth_default_user = web_auth_default_user
 
         assert len(self.masters) == len(self.public_masters)
 
     def get_user_session(self, user):
-        """Returns a copy of self with auth headers set for user
+        """Returns a copy of this client but with auth for user (can be None)
         """
-        new_session = copy.deepcopy(self)
-        # purge old auth headers
-        if self.web_auth_default_user is not None:
-            for k in self.web_auth_default_user.auth_header.keys():
-                if k in new_session.default_headers:
-                    del new_session.default_headers[k]
-        # if user is given then auth and update the headers
-        new_session.web_auth_default_user = user
+        new = self.get_client()
+        new.session.auth = None
+        new.session.cookies.clear()
```

Comment:
Clear is required because I think we will get auth rejection if we send requests from a session that has cookies set from a previous user while supplying an Authorization header for the new user